tuist/tuist
# Add metal diagnostics options to 'RunAction'

### Short description 📝

Add ability to set metal diagnostic options from Tuist 'Scheme' configuration to be more precise in 'RunAction'
Example of code:
```swift
let project = Project(
    // ...
    schemes: [
        .scheme(
            name: "MyApp",
            buildAction: .buildAction(targets: [.target("MyApp")]),
            runAction: .runAction(
                metalOptions: .options(
                    showGraphicsOverview: true
                )
            )
        )
    ],
    // ...
)
```
Equivalent in xcode:
![Screenshot](https://github.com/user-attachments/assets/d24dd2da-d225-4706-bbbe-d19cee601d88)

There would be 2 additional pull request to [tuist/XcodeGraph](https://github.com/tuist/XcodeGraph/pull/21) and [tuist/XcodeProj](https://github.com/tuist/XcodeProj/pull/828)

### How to test the changes locally 🧐

Clone the repository from "git clone https://github.com/filipracki/tuist.git"
Checkout the branch "git checkout add-metal-diagnostics-options"
Then follow the documentation (see [the documentation](https://docs.tuist.io/contributors/get-started) for reference).

### Contributor checklist ✅


### Reviewer checklist ✅
